src/fsharp/FSharp.Core/control.fs

```diff
@@ -1541,8 +1541,8 @@ namespace Microsoft.FSharp.Control
             let continuation (completedTask : Task<_>) : unit =
                 args.aux.trampolineHolder.Protect((fun () ->
                     if completedTask.IsCanceled then
-                        if useCcontForTaskCancellation then args.aux.ccont(new OperationCanceledException())
-                        else args.aux.econt (ExceptionDispatchInfo.Capture(new TaskCanceledException()))
+                        if useCcontForTaskCancellation then args.aux.ccont(new OperationCanceledException(args.aux.token))
```

Comment:
@saul @matthid this line should be reverted to the original (unless there is a generic way to obtain a cancellation token from arbitrary tasks which I'm not aware of)